### PR TITLE
refactor: apply new rewards rules PE-1688

### PR DESCRIPTION
**Pull request by karlprieb**

### CODE REVIEW

**Review by matibat — APPROVED** [review]
Looks good to me. Thank you!

File: src/constants.ts [hunk 1]
```diff
@@ -3,12 +3,15 @@ import { WalletStatEntry } from './inferno_types';
...
+export const ONE_HUNDRED_MB = 1000 * 1000 * 100;
+export const GROUP_EFFORT_REWARDS_FULL = [
 	75, 55, 50, 47, 45, 43, 40, 37, 35, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13,
 	12, 11, 10, 9, 9, 8, 8, 7, 7, 6, 6, 5, 5, 4, 4, 3, 3, 2, 2, 1, 1
 ] as const;
-export const GQL_URL = 'https://arweave.net/graphql';
+export const GROUP_EFFORT_REWARDS_HALF = [
+	38, 28, 25, 24, 23, 22, 20, 19, 18, 16, 16, 15, 14, 14, 13, 13, 12, 12, 11, 11, 10, 10, 9, 9, 8, 8, 7, 7, 6, 6, 5,
+	5, 5, 4, 4, 4, 4, 3, 3, 3, 3, 2, 2, 2, 2, 1, 1, 1, 1, 1
+] as const;
```
Comment (matibat) [inline src/constants.ts]:
```ts
for (const rewardIndex in GROUP_EFFORT_REWARDS_FULL) {
    const reward = GROUP_EFFORT_REWARDS_FULL[rewardIndex];
    const expectedHalfReward = Math.ceil(reward / 2);
    const halfReward = GROUP_EFFORT_REWARDS_HALF[rewardIndex];
    if (expectedHalfReward !== halfReward) {
        console.log(`Missmatch at index ${rewardIndex} (${reward}): expected:${expectedHalfReward} => actual:${halfReward}`);
 }
}

/**
Missmatch at index 12 (29): expected:15 => actual:14
Missmatch at index 14 (27): expected:14 => actual:13
Missmatch at index 16 (25): expected:13 => actual:12
Missmatch at index 18 (23): expected:12 => actual:11
Missmatch at index 20 (21): expected:11 => actual:10
Missmatch at index 22 (19): expected:10 => actual:9
Missmatch at index 24 (17): expected:9 => actual:8
Missmatch at index 26 (15): expected:8 => actual:7
Missmatch at index 28 (13): expected:7 => actual:6
Missmatch at index 30 (11): expected:6 => actual:5
Missmatch at index 33 (9): expected:5 => actual:4
Missmatch at index 37 (7): expected:4 => actual:3
Missmatch at index 41 (5): expected:3 => actual:2
Missmatch at index 45 (3): expected:2 => actual:1
**/
```
There are some inconsistencies here, we actually want the output of this:
```ts
GROUP_EFFORT_REWARDS_FULL.map(reward => Math.ceil(reward / 2));
```

```suggestion
export const GROUP_EFFORT_REWARDS_HALF = [
	38, 28, 25, 24, 23, 22, 20, 19, 18, 16, 16, 15, 15, 14, 14, 13, 13, 12, 12, 11, 11, 10, 10, 9, 9, 8, 8, 7, 7, 6, 6,
	5, 5, 5, 4, 4, 4, 4, 3, 3, 3, 3, 2, 2, 2, 2, 1, 1, 1, 1
] as const;
```

Comment (matibat) [reply src/constants.ts]: We can either have it hardcoded as it is, we can have the mapping, or/and we can have a test to test for it

File: src/daily_output.ts [hunk 2]
```diff
@@ -155,7 +135,13 @@ export class DailyOutput {
 			.map((address, index) => {
 				const rankPosition = index + 1;
 				const isInTop50 = rankPosition <= 50;
-				const rewards = hasReachedMinimumGroupEffort && isInTop50 ? GROUP_EFFORT_REWARDS[index] : 0;
+				let rewards;
+				if (hasReachedMinimumGroupEffort) {
```
Comment (matibat) [inline src/daily_output.ts:139]: Nice! Now we do have rewards filled up even if there are less than 50 wallets 👏

File: src/utils/download_output.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { GQL_URL, gatewayUrl } from '../constants';
```
Comment (matibat) [inline src/utils/download_output.ts:1]: We can have a tech_debt to implement retries in these requests

File: src/constants.ts (hunk 1 above)
Comment (matibat) [reply src/constants.ts]: Any of both suggestions are good to me
```suggestion
export const GROUP_EFFORT_REWARDS_HALF = GROUP_EFFORT_REWARDS_FULL.map(reward => Math.ceil(reward / 2));
```

**Review by matibat — CHANGES_REQUESTED** [review]
See my comments. Nice improvements overall, I liked the cleanup 👏

File: src/constants.ts (hunk 1 above)
Comment (karlprieb) [reply src/constants.ts]: I'll check the requirements again and make a decision.
Thank you!

File: src/utils/download_output.ts (hunk 3 above)
Comment (karlprieb) [reply src/utils/download_output.ts:1]: I don't think it's worth it. It's just a couple calls that you can retry running the script again.

Comment (matibat) [reply src/utils/download_output.ts:1]: Exactly! It's not worth it at the moment, we can have that as a tech_debt ticket

File: src/constants.ts (hunk 1 above)
Comment (matibat) [reply src/constants.ts]: Nice

**Review by matibat — APPROVED** [review]
Looks good to me now. Thanks!

**Pull request merged**